pokemongo_bot/tree_config_builder.py

```diff
@@ -27,7 +27,7 @@ def build(self):
             task_config = task.get('config', {})
 
             worker = self._get_worker_by_name(task_type)
-            instance = worker(self.bot, task_config)
+            instance = worker(self.bot, task_config, task_type)
```

Comment:
Why do we want to pass in the name of the task to the task itself? It already knows. It's its class name